CA integration through Envoy SDS

maxlambrecht opened the pull request:
**Please provide a description of this PR:**

This PR includes the following changes: 

1. Make `istio-agent` check the presence of a Unix Domain Socket on the fixed path `/var/run/secrets/workload-spiffe-uds/socket`:

           1. In the case where the socket exists, istio-agent does not start its own SDS Server.
           2. In the case where the socket does not exist, istio-agent checks whether certificate files on the fixed path folder (/var/run/secrets/workload-spiffe-credentials) exists.  If the certificate files exist, istio-agent starts its own SDS Server, listening on the fixed socket path (/var/run/secrets/workload-spiffe-uds/socket); to serve those certificate files, keeping file  watchers on them. Envoy proxy connects to the SDS Server run by istio-agent through the fixed socket path and gets the cryptographic materials of the certificate files served by the SDS API.
           3. In case istio-agent does not find either the socket, or the certificate files in the fixed path;  it starts its own SDS Server using a `caClient` to connect to istiod to get the cryptographic materials (See CA Flow).

 Added new tests for istio-agent covering the scenarios where the socket is detected and where the certificates are detected.

2.  Added volumes and volumeMounts for the fixed path to the chart manifests. 

3. Updated all the affected testdata files. 

4. Update agent README with the details of the new CA integration approach.

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [x] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [X] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

maxlambrecht committed:
CA integration through SDS:
 - Make agent detect socket and certs files on fixed paths.
 - Envoy bootstrap template points to fixed socket path.
 - Unit tests.

Signed-off-by: Max Lambrecht <maxlambrecht@gmail.com>

maxlambrecht committed:
CA integration through SDS:
 - Make agent detect socket and certs files on fixed paths.
 - Envoy bootstrap template points to fixed socket path.
 - Unit tests.

Signed-off-by: Max Lambrecht <maxlambrecht@gmail.com>

maxlambrecht committed:
CA integration through SDS:
- Add workload-socket and workload-certs volumes and mounts to manifest charts.

Signed-off-by: Max Lambrecht <maxlambrecht@gmail.com>

maxlambrecht committed:
CA integration through SDS:
- Add workload-socket and workload-certs volumes and mounts to manifest charts.

Signed-off-by: Max Lambrecht <maxlambrecht@gmail.com>

maxlambrecht committed:
CA integration through SDS:
- Add workload-socket and workload-certs volumes and mounts to testdata files.

Signed-off-by: Max Lambrecht <maxlambrecht@gmail.com>

maxlambrecht committed:
CA integration through SDS:
- Add workload-socket and workload-certs volumes and mounts to testdata files.

Signed-off-by: Max Lambrecht <maxlambrecht@gmail.com>

maxlambrecht committed:
CA integration through SDS:
- Adding details to Agent README.

Signed-off-by: Max Lambrecht <maxlambrecht@gmail.com>

maxlambrecht committed:
CA integration through SDS:
- Adding details to Agent README.

Signed-off-by: Max Lambrecht <maxlambrecht@gmail.com>

deveshkandpal1224 committed:
Configure warmupDurationSecs for slow start mode (#37583)

* Configure warmupDurationSecs for slow start mode

* pass entire loadbalancer

* Add additional test for rr / lr when warmup not set

* fix test

howardjohn committed:
Partially revert changes to image building (#37548)

Turns out we use operator more than I thought in postsubmit. Also set
default values for these vars in case we run outside prow

ramaraochavali committed:
fix passing exit on zero active connections to metadata (#37573)

* fix passing exit on zero active connections to metadata

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* add statsoption test

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

ramaraochavali committed:
remove proxy 112 checks (#37575)

* remove proxy 112 checks

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* fix tests

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* dead code

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* remove correct dead code

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* remove correct dead code

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* fix ut

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

SpecialYang committed:
Enable validation for mcp-over-xds store (#37579)

istio-testing committed:
Automator: update common-files@master in istio/istio@master (#37588)

litong01 committed:
Make analysis interval configurable (#37505)

* Make analysis interval configurable

Signed-off-by: Tong Li <litong01@us.ibm.com>

* Make analysis interval configurable

Signed-off-by: Tong Li <litong01@us.ibm.com>

* Make analysis interval configurable

Signed-off-by: Tong Li <litong01@us.ibm.com>

istio-testing committed:
Automator: update istio/client-go@master dependency in istio/istio@master (#37590)

istio-testing committed:
Automator: update proxy@master in istio/istio@master (#37595)

istio-testing committed:
Automator: update istio/client-go@master dependency in istio/istio@master (#37600)

howardjohn committed:
Disable Envoy retries in tests (#37598)

istio-testing committed:
Automator: update common-files@master in istio/istio@master (#37599)

nmittler committed:
[TF] Removing RBAC test utiities (#37549)

This is removing one of the extra testing frameworks from the security tests. Preferring to use the raw echo libraries to the extent possible.

dddddai committed:
Support REPLACE for virtual host in envoy filter (#36582)

* support replace for virtual host in envoy filter

* rebase

hzxuzhonghu committed:
Set upstream max tls version to TLS 1.3 (#37540)

* Set max tls version to 1.3 for upstream

* Fix ut

* add release note

* fix

hzxuzhonghu committed:
Clone catch all vhost to prevent it from being mutated (#37612)

Commit:
fix a bug that cache entries were never updated (#37578)

* fix a bug that cache entries were never updated

* change cacheEntry to point

* add UT

howardjohn committed:
Pilot debounce: log reason when there is no config (#37603)

ichbinfrog committed:
Reduce test duration for google mock CA (#37597)

* Reduce sleep delay for mock CA server

Signed-off-by: Trung <trung.hoang@pricehubble.com>

* Reduce sleep duration for google-cas mock

Signed-off-by: Trung <trung.hoang@pricehubble.com>

* Remove sleep safeguard

Signed-off-by: Trung <trung.hoang@pricehubble.com>

ramaraochavali committed:
increase test timeout (#37615)

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

istio-testing committed:
Automator: update proxy@master in istio/istio@master (#37627)

Commit:
Update to split linux for building on Macs (#37633)

* Update to split linux for building on Macs

* Fix imports

nmittler committed:
[TF] Refactoring ConfigManager interface (#37628)

Cleaner support for evaluating templates, reading from files, and waiting for apply to complete.

nmittler committed:
[TF] Fixing telemetry test nesting (#37636)

The telemetry tests are incorrectly using subtests. This cleans things up and avoids this confusion in the future by renaming all the `testing.T` and `framework.TestContext` variables to `t` in order to allow scoping to choose the correct one.

bianpengyuan committed:
Make include outbound port work with CNI. (#37639)

* Make include outbound port work with CNI.

* rn

howardjohn committed:
Speed up TestK8sSignWithMeshConfig (#37640)

howardjohn committed:
Restrict another hub to prevent accidents (#37591)

istio-testing committed:
Automator: update proxy@master in istio/istio@master (#37650)

Commit:
oci wasm module's cache key use image's digest instead of binary's checksum (#37566)

hzxuzhonghu committed:
Revert cache catch all virtualhost (#37648)

* Revert "Clone catch all vhost to prevent it from being mutated (#37612)"

This reverts commit e60eee9448444f5f087c0877f02549f10e3429b3.

* Revert "do not build catch all virtual host every time (#35449)"

This reverts commit b32710c73525c0ba0df8af1d4903d947da68e178.

* fix build

dddddai committed:
fix: mcs incorrectly deleted (#37624)

Signed-off-by: dddddai <dddwq@foxmail.com>

ramaraochavali committed:
skip active connections check when downstream cx stats are not available (#37559)

* skip active connections check when downstream cx stats are not available

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* move to warn

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* fix error message

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* hanndle error differently

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* lint

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

hzxuzhonghu committed:
optimize: auto allocate addresses only when services updated (#37616)

* optimize: auto allocate addresses only when services updated

* unlock ASAP

* shallow copy svc as only need to copy AutoAllocatedAddress

* skip govet

Commit:
update DerivedGauge (#37653)

* update DerivedGauge

* fix gen & lint

howardjohn committed:
Log remaining names in operator test (#37662)

Commit:
Remove default caBundle value from Chart to allow GitOps (#37655)

* Remove default caBundle value from Chart to allow GitOps

* Remove default caBundle value from Chart to allow GitOps

* Remove default caBundle value from Chart to allow GitOps

jufantozzi commented on pkg/istio-agent/README.md:
```diff
@@ -14,6 +14,20 @@ requests to the agent, which will be forwarded to the configured discovery serve
 
 ## CA Flow
 
+istio-agent checks the presence of a socket file on the well-known path `/var/run/secrets/workload-spiffe-uds/socket`
```
One thing I was also considering was bolding 'socket path' and 'certificate path' to help ease reading. I wont add this to every suggestion but wanted to throw the idea 😛 
If you like it or not let me know so I can update accordingly.
```suggestion
istio-agent checks the presence of a socket file on the defined **socket path** `/var/run/secrets/workload-spiffe-uds/socket`.
```

jufantozzi commented on pkg/istio-agent/README.md:
```diff
@@ -14,6 +14,20 @@ requests to the agent, which will be forwarded to the configured discovery serve
 
+istio-agent checks the presence of a socket file on the well-known path `/var/run/secrets/workload-spiffe-uds/socket`
+
+1. In the case where the socket exists, istio-agent does not start its own SDS Server.
```
```suggestion
1. If a socket is found, istio-agent will not start its own SDS Server and Envoy will be configured to use that socket as its source of cryptographic material.
```

jufantozzi commented on pkg/istio-agent/README.md:
```diff
@@ -14,6 +14,20 @@ requests to the agent, which will be forwarded to the configured discovery serve
+istio-agent checks the presence of a socket file on the well-known path `/var/run/secrets/workload-spiffe-uds/socket`
+
+1. In the case where the socket exists, istio-agent does not start its own SDS Server.
+2. In the case where the socket does not exist, istio-agent checks whether the certificate files are present on the well-known path folder `/var/run/secrets/workload-spiffe-credentials`.
```
```suggestion
2. If a socket is not found, istio-agent then checks whether certificate files are present or not on the defined certificate path `/var/run/secrets/workload-spiffe-credentials`. 
```

jufantozzi commented on pkg/istio-agent/README.md:
```diff
@@ -14,6 +14,20 @@ requests to the agent, which will be forwarded to the configured discovery serve
+
+1. In the case where the socket exists, istio-agent does not start its own SDS Server.
+2. In the case where the socket does not exist, istio-agent checks whether the certificate files are present on the well-known path folder `/var/run/secrets/workload-spiffe-credentials`.
+   If the certificate files exist, istio-agent starts its own SDS Server, listening on the fixed socket path `/var/run/secrets/workload-spiffe-uds/socket`;
```
```suggestion
   If certificate files are found, istio-agent will start its own SDS Server, listening and serving these certificates on the defined socket path `/var/run/secrets/workload-spiffe-uds/socket`,
```

howardjohn committed:
gateway: avoid recreating Deployment on each kubectl apply (#37670)

* gateway: avoid recreating Deployment on each kubectl apply

Currently, kubectl apply annotation from Gateway is propogated to
Deployment which triggers a restart everytime we change any field.

* Update templates.go

jufantozzi commented on pkg/istio-agent/README.md:
```diff
@@ -14,6 +14,20 @@ requests to the agent, which will be forwarded to the configured discovery serve
+1. In the case where the socket exists, istio-agent does not start its own SDS Server.
+2. In the case where the socket does not exist, istio-agent checks whether the certificate files are present on the well-known path folder `/var/run/secrets/workload-spiffe-credentials`.
+   If the certificate files exist, istio-agent starts its own SDS Server, listening on the fixed socket path `/var/run/secrets/workload-spiffe-uds/socket`;
+   to serve those certificate files, keeping file watchers on them. Envoy proxy connects to the SDS Server run by istio-agent
```
```suggestion
   while also keeping file watchers on them. Envoy proxy then connects to istio-agent's SDS Server
```

nmittler committed:
[TF] Remove security Checker (#37358)

This removes another major layer within the security tests.

howardjohn committed:
Remove duplicate test case (#37668)

bianpengyuan committed:
Add fake container registry component for Wasm pulling integration test. (#37601)

* Add fake container registry component for Wasm pulling integration test.

* update.

jufantozzi commented on pkg/istio-agent/README.md:
```diff
@@ -14,6 +14,20 @@ requests to the agent, which will be forwarded to the configured discovery serve
+2. In the case where the socket does not exist, istio-agent checks whether the certificate files are present on the well-known path folder `/var/run/secrets/workload-spiffe-credentials`.
+   If the certificate files exist, istio-agent starts its own SDS Server, listening on the fixed socket path `/var/run/secrets/workload-spiffe-uds/socket`;
+   to serve those certificate files, keeping file watchers on them. Envoy proxy connects to the SDS Server run by istio-agent
+   through the fixed socket path and gets the cryptographic materials of the certificate files served by the SDS API.
```
```suggestion
   through the defined socket path and gets the cryptographic materials of the certificate files served by the SDS API.
```

maxlambrecht replied:
I would use 'well-known' instead of 'fixed', ah but you suggested changing 'well-know' by defined as well, I think 'defined' is good 😃

jufantozzi commented on pkg/istio-agent/README.md:
```diff
@@ -14,6 +14,20 @@ requests to the agent, which will be forwarded to the configured discovery serve
+   If the certificate files exist, istio-agent starts its own SDS Server, listening on the fixed socket path `/var/run/secrets/workload-spiffe-uds/socket`;
+   to serve those certificate files, keeping file watchers on them. Envoy proxy connects to the SDS Server run by istio-agent
+   through the fixed socket path and gets the cryptographic materials of the certificate files served by the SDS API.
+3. In case istio-agent does not find either the socket, or the certificate files in the fixed path
```
```suggestion
3. If istio-agent does not find either the socket, or the certificate files in their respective paths
```

jufantozzi commented on pkg/istio-agent/README.md:
```diff
@@ -14,6 +14,20 @@ requests to the agent, which will be forwarded to the configured discovery serve
+   to serve those certificate files, keeping file watchers on them. Envoy proxy connects to the SDS Server run by istio-agent
+   through the fixed socket path and gets the cryptographic materials of the certificate files served by the SDS API.
+3. In case istio-agent does not find either the socket, or the certificate files in the fixed path
+   it starts its own SDS Server using a `caClient` to connect to istiod to get the cryptographic materials (See Default CA Flow).
```
```suggestion
   it will start its own SDS Server using a `caClient` to connect to istiod and to fetch cryptographic materials (See Default CA Flow).
```

Commit:
Update --no-clobber to not check existence of tag `latest` (#37669)

* Remove --no-clobber since we want to overwrite :latest

* Ignore latest when using --no-clobber

istio-testing committed:
Automator: update proxy@master in istio/istio@master (#37676)

Commit:
pilot-agent add cert expiry metric (#37372)

* pilot-agent add cert expiry metric

* fix gen & lint

howardjohn committed:
retry: fix default backoff (#37646)

Before we set max to the base. Instead set it to 16*base, just like
explicit calls

Commit:
Make all telemetry tests support multiple clusters (#37514)

hzxuzhonghu committed:
build default transport socketmatch each time (#37678)

jufantozzi reviewed (approved):
Great work Max! this looks pretty tidy and clean, well done!

litong01 committed:
Fixed #37761 (#37684)

Signed-off-by: Tong Li <litong01@us.ibm.com>

Commit:
Apply suggestions to README.

Co-authored-by: Juliano Fantozzi <juliano.fantozzi@gmail.com>

Commit:
Apply suggestions to README.

Co-authored-by: Juliano Fantozzi <juliano.fantozzi@gmail.com>

zirain committed:
update metric description (#37683)

howardjohn committed:
Pilot: store latest push instead of global (#37604)

* Pilot: store latest push instead of global

* Fix some more usages and add comments

jufantozzi commented on pkg/istio-agent/README.md:
```diff
@@ -14,6 +14,20 @@ requests to the agent, which will be forwarded to the configured discovery serve
+istio-agent checks the presence of a socket file on the defined **socket path** `/var/run/secrets/workload-spiffe-uds/socket`.
+
+1. If a socket is found, istio-agent will not start its own SDS Server and Envoy will be configured to use that socket as its source of cryptographic material.
+2. If a socket is not found, istio-agent then checks whether certificate files are present or not on the defined certificate path `/var/run/secrets/workload-spiffe-credentials`. 
```
```suggestion
2. If a socket is not found, istio-agent then checks whether certificate files are present or not on the defined **certificate path** `/var/run/secrets/workload-spiffe-credentials`. 
```

jufantozzi commented on pkg/istio-agent/README.md:
```diff
@@ -14,6 +14,20 @@ requests to the agent, which will be forwarded to the configured discovery serve
+
+1. If a socket is found, istio-agent will not start its own SDS Server and Envoy will be configured to use that socket as its source of cryptographic material.
+2. If a socket is not found, istio-agent then checks whether certificate files are present or not on the defined certificate path `/var/run/secrets/workload-spiffe-credentials`. 
+   If certificate files are found, istio-agent will start its own SDS Server, listening and serving these certificates on the defined socket path `/var/run/secrets/workload-spiffe-uds/socket`,
```
```suggestion
   If certificate files are found, istio-agent will start its own SDS Server, listening and serving these certificates on the defined **socket path** `/var/run/secrets/workload-spiffe-uds/socket`,
```

jufantozzi reviewed (commented):
For consistency

howardjohn committed:
mitigate flaky tests temporarily (#37672)

* mitigate flaky tests temporarily

TestClusterLocal has a known fix which will land within a week.

The other one does not; we are not skipping though just making the test
slightly less exhaustive

* Update multicluster_test.go

mchurichi commented on pkg/bootstrap/testdata/all_golden.json:
```diff
@@ -338,7 +338,7 @@
                 "address":{
                   "pipe": {
-                    "path": "/tmp/bootstrap/all/SDS"
+                    "path": "./var/run/secrets/workload-spiffe-uds/socket"
```
is there a reason why these file paths are now relatives?

maxlambrecht replied:
> is there a reason why these file paths are now relatives?

The defined fixed socket path is relative: `WorkloadIdentitySocketPath = "./var/run/secrets/workload-spiffe-uds/socket"`, there are several examples of relatives paths in istio and it was also suggested by one of the maintainers. Also the path in the Envoy bootstrap is relative and thus all these `testadata` files need to have the same relative path otherwise the tests fail.

mchurichi commented on pkg/istio-agent/agent.go:
```diff
@@ -405,13 +405,30 @@ func (a *Agent) Run(ctx context.Context) (func(), error) {
-		return nil, fmt.Errorf("failed to start workload secret manager %v", err)
-	}
+	if socketFileExists(security.WorkloadIdentitySocketPath) {
+		log.Info("SDS socket detected, don't start own SDS Server")
```
how about something like `SDS socket found. Istio SDS Server won't be started`? to keep messages similar

mchurichi commented on pkg/istio-agent/agent.go:
```diff
@@ -405,13 +405,30 @@ func (a *Agent) Run(ctx context.Context) (func(), error) {
 
-	a.sdsServer = sds.NewServer(a.secOpts, a.secretCache)
-	a.secretCache.SetUpdateCallback(a.sdsServer.UpdateCallback)
+		log.Info("SDS socket not detected, starting own SDS Server")
```
Suggestion: `SDS socket not found. Starting Istio SDS Server`

mchurichi commented on pkg/istio-agent/agent_test.go:
```diff
@@ -259,6 +261,47 @@ func TestAgent(t *testing.T) {
+		secOpts := &security.Options{}
+		secOpts.RootCertFilePath = dir + "/root-cert.pem"
+		secOpts.CertChainFilePath = dir + "/cert-chain.pem"
+		secOpts.KeyFilePath = dir + "/key.pem"
```
nit: shouldn't we use `filepath.Join(dir, "xxxx")` to keep the same conventions used in the rest of the tests?

howardjohn committed:
Fix release builder fetch (#37696)

* Fix release builder fetch

* Fix license mirror

* Fix security test

* Regenerate certs

nmittler committed:
[TF] General echo cleanup (#37693)

Splitting out structures into separate files and creating substructures in ConfigOptions.

Commit:
Fix --no-clobber ignoring of `latest` images (#37697)

istio-testing committed:
Automator: update istio/api@master dependency in istio/istio@master (#37706)

howardjohn committed:
Add script to build kind node image (#37688)

Now that we will do multi-arch builds its a bit more complex than a
single command, so check in a script to make it easy

istio-testing committed:
Automator: update common-files@master in istio/istio@master (#37690)

howardjohn committed:
Improve gateway chart docs (#37659)

istio-testing committed:
Automator: update istio/client-go@master dependency in istio/istio@master (#37692)

istio-testing committed:
Automator: update common-files@master in istio/istio@master (#37712)

nmittler committed:
[TF] Move retry options to CallOptions (#37707)

Fixes #37686

Commit:
Apply suggestions to README.

Co-authored-by: Juliano Fantozzi <juliano.fantozzi@gmail.com>

Commit:
Apply suggestions to README.

Co-authored-by: Juliano Fantozzi <juliano.fantozzi@gmail.com>

istio-testing committed:
Automator: update istio/client-go@master dependency in istio/istio@master (#37719)

maxlambrecht committed:
Apply suggestion to agent.go and agent_test.go.

Signed-off-by: Max Lambrecht <maxlambrecht@gmail.com>

maxlambrecht committed:
Apply suggestion to agent.go and agent_test.go.

Signed-off-by: Max Lambrecht <maxlambrecht@gmail.com>

maxlambrecht committed:
Merge branch 'sds-integration' of https://github.com/maxlambrecht/istio into sds-integration

hanxiaop committed:
Fix describe po not finding vs when protocol is https (#36713)

* fix

* add releasenote

* add integration test

* fix test

* unit test

howardjohn committed:
Optimize injection and operator jobs (#37607)

* Optimize injection and operator jobs

For https://github.com/istio/istio/issues/37552

* Precompute

* Optimize operator tests

* Filter down rendered manifests

* fix

* lint

* fix

nmittler committed:
[TF] Rename echoboot to deployment. (#37727)

Clarifies the purpose of the Builder and makes the call sites more obvious.

douglas-reid committed:
TelemetryAPI: Disable Stackdriver logging if not configured (by default, or otherwise)  (#37708)

* TelemetryAPI: Disable Stackdriver logging if not configured

* lint and release notes

jacob-delgado committed:
UPDATE_BRANCH=master ./bin/update_deps.sh (#37737)

istio-testing committed:
Automator: update common-files@master in istio/istio@master (#37745)

istio-testing committed:
Automator: update istio/client-go@master dependency in istio/istio@master (#37746)

howardjohn committed:
Fix issue with helm tags rendering (#37740)

* Fix issue with helm tags rendering

Regression since
https://github.com/istio/istio/commit/098cc47dfcbdca202208bd241d9dee35cd230a4d

Apparently the `define` here is not actually used since its already
define

* add template back

istio-testing committed:
Automator: update proxy@master in istio/istio@master (#37752)

ramaraochavali committed:
check key and cert are valid before sending to Envoy (#37722)

* check key and cert are valid before sending to Envoy

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* fix tests

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* fix tests

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

markmc committed:
gateway: fix error log when deleting a Gateway (#37730)

Removes error logs like:

  error   gateway unable to fetch Gateway: gateway.gateway.networking.k8s.io "gateway" not found  gateway=istio-ingress/gateway

Seems the intent here is to ignore not found errors.

nmittler committed:
[TF] More echo cleanup (#37744)

- Removing WorkloadPort and just adding to the Ports list.
- Updated common ports to be a single list that includes both service and workload-only ports.
- Renaming CallOptions.Target->To. Just a bit more concise.
- Added Ports utility and moving aggregate port operations there.
- Added Configurable and WorkloadContainer interfaces in preparation for allowing different target types (e.g. Instances).
- Added Cluster to Workload in preparation to switching to use workloads for accessing all clusters for a set of Instances.

istio-testing committed:
Update BASE_VERSION to master-2022-03-05T19-01-04 (#37760)

howardjohn committed:
Switch protobuf compiling to `buf` (#37738)

* Switch protobuf compiling to `buf`

This matches what we have in istio/api, cleans up some really ugly
makefiles, enrolls `echo.proto` in automation (it is manual today), is
faster, and makes it trivial to switch compilers in the future (off
gogo)

* regen grpc

howardjohn committed:
Add tests for virtual services (#37747)

howardjohn committed:
Revert "Pilot: store latest push instead of global (#37604)" (#37759)

This reverts commit 6b93316d353d9e00f7ecd23ab5c83f9ec91c5a74.

howardjohn committed:
Standardize run commands for integ tests (#37761)

Some did not have `-vet=false`

istio-testing committed:
Automator: update proxy@master in istio/istio@master (#37763)

ramaraochavali committed:
fix return in agent code (#37770)

* fix return in agent code

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* lint

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

ramaraochavali committed:
remove unused functions (#37768)

* remove unused functions

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* use same type

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* format

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

hanxiaop committed:
improve config store error return (#37774)

douglas-reid committed:
telemetry api: allow default json format for envoy file logger (#37733)

* telemetry api: allow default json format for envoy file logger

* lint fix

istio-testing committed:
Automator: update proxy@master in istio/istio@master (#37782)

hanxiaop committed:
clean up unused code (#37775)

howardjohn committed:
Make AuthorizationPolicy parsing more robust (#37777)

howardjohn committed:
go dependencies: bump protobuf library (#37784)

howardjohn committed:
pilot: limit reads from untrusted inputs (#37785)

* pilot: limit reads from untrusted inputs

* revert jwks

howardjohn committed:
Cleanup stale file (#37783)

howardjohn committed:
Add new `go-compile-verbose` tool to help debug build times (#37780)

* Add new `go-compile-verbose` tool to help debug build times

* fix lint issues

alexandrealvino reviewed (approved):
Well done 🥇

istio-testing committed:
Automator: update proxy@master in istio/istio@master (#37807)

nmittler committed:
[TF] Cleanup echo Port (#37808)

We currently have Port and PortName and the former is a pointer. This leads to a lot of complexity to the logic. This PR removes PortName and makes Port a struct rather than a pointer.

costinm committed:
add RBAC support to proxyless gRPC (#37076)

* Add RBAC support for proxyless grpc

Change-Id: I78d72ed96d559645e8b59f272c80c25ad13b9696

* Minor refactoring of the test

Change-Id: Idc2bad396aa9fb5d77716dcc7272c07b406086e4

* Use unique names

Change-Id: I36854fa6245b6c674b96cbef9eb9e6106f3cc217

* Update grpc after merge

Change-Id: I35ba19c01d28066dee3fa756abb942f0eb2d8493

* Fix merge

Change-Id: Ia6d7d2fa54f6bcea0d6985ce36cb51e4ba77f752

* Lints

Change-Id: Ib072e7a9b3544b70fc1456c0c4557702019495f1

* gen

Change-Id: Ia254fe33a7f80cd1c620d8b1dab06f73263985ea

Co-authored-by: Costin Manolache <costin@google.com>

hzxuzhonghu committed:
Enable serviceentry wles cross clusters (#37675)

howardjohn committed:
Do not waste time compressing VM packages in tests (#37811)

istio-testing committed:
Automator: update common-files@master in istio/istio@master (#37810)

nmittler committed:
[TF] Flexible echo CallOptions.To (#37781)

Creates a `Target` interface that is implemented by both `Instance` and `Instances`.

Cleaning up call sites that explicitly pass in `Instance` to now pass in `Instances` where possible. Check for `ReachedClusters` now operates directly on `CallOptions`.

howardjohn committed:
Pilot: store latest push instead of global (#37800)

* Revert "Revert "Pilot: store latest push instead of global (#37604)" (#37759)"

This reverts commit f6b1aa2d1956712018cd69051a7405424fbb7e04.

* Use last push time for push requests

Fixes https://github.com/istio/istio/issues/37709

(cherry picked from commit 171f74f7aa10479d8c3477e111d7f536853a8a6a)

* fix lint

Commit:
add otel provider in demo profile (#37819)

* add otel provider in demo profile

* update script

istio-testing committed:
Automator: update proxy@master in istio/istio@master (#37827)

ramaraochavali committed:
minor dns fix and additional tests (#37833)

* minor dns fix and additional tests

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* fix test

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

nmittler committed:
[TF] Removing some multi-cluster specific logic (#37302)

ramaraochavali committed:
fix test file secrets flake (#37826)

* fix testfilesecrets flake

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* Update security/pkg/nodeagent/cache/secretcache_test.go

Co-authored-by: John Howard <howardjohn@google.com>

ycai-aspen committed:
add dual stack support for echo server (#37817)

nmittler committed:
Synchronize MCS even handling (#37839)

The MCS serviceImportCacheImpl.onServiceEvent is called concurrently from every service update in the mesh. This logic needs to be processed synchronously as a unit to avoid races.

This PR moves the processing for this method onto the local cluster's queue.

howardjohn committed:
Move managed gateway our of loop (#37837)

Currently its inside the Ingresses loop; but we are deploying a stand
alone gateway, so it doesn't make sense to deploy it 3 times

Commit:
fix nit (#37845)

litong01 committed:
Fixed metallb master branch issue (#37849)

latest metallb master branch is broken. Switch to use the latest
working version v0.12

Signed-off-by: Tong Li <litong01@us.ibm.com>

ramaraochavali committed:
fix data race in dns (#37852)

* minor dns fix and additional tests

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* fix test

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* add test

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

istio-testing committed:
Automator: update proxy@master in istio/istio@master (#37858)

howardjohn committed:
Optimize single node kind cluster (#37862)

* Optimize single node kind cluster

Recommended by https://github.com/kubernetes/minikube/issues/5682

* Update prow/config/default.yaml

Co-authored-by: Eric Van Norman <ericvn@us.ibm.com>

Co-authored-by: Eric Van Norman <ericvn@us.ibm.com>

maxlambrecht committed:
Merge branch 'sds-integration' of https://github.com/maxlambrecht/istio into sds-integration

maxlambrecht commented:
Closing this PR, the commits were squashed and copied to a clean branch to submit to upstream istio repo.
